notifications_utils/request_helper.py

```diff
@@ -45,3 +45,47 @@ def rewrite_response_headers(status, headers, exc_info=None):
 def init_app(app):
     app.request_class = CustomRequest
     app.wsgi_app = ResponseHeaderMiddleware(app.wsgi_app, 'NotifyRequestID')
+
+
+def check_proxy_header_before_request():
+    if current_app.config['DEBUG']:
+        return None
+
+    keys = [
+        current_app.config.get('ROUTE_SECRET_KEY_1'),
+        current_app.config.get('ROUTE_SECRET_KEY_2'),
+    ]
+    result, msg = _check_proxy_header_secret(request, keys)
+
+    if not result:
+        current_app.logger.warning(msg)
+        abort(403)
+
+    # We need to return None to continue processing the request
+    # http://flask.pocoo.org/docs/0.12/api/#flask.Flask.before_request
+    current_app.logger.info(msg)
+    return None
+
+
+def _check_proxy_header_secret(request, secrets, header='X-Custom-Forwarder'):
+    if header not in request.headers:
+        return False, "Header missing"
+
+    header_secret = request.headers.get(header)
+    if not header_secret:
+        return False, "Header exists but is empty"
+
+    # if there isn't any non-empty secret configured we fail closed
+    if not any(secrets):
+        return False, "Secrets are not configured"
+
+    key_used = None
+    for i, secret in enumerate(secrets):
+        if header_secret == secret:
+            key_used = i + 1  # add 1 to make it human-compatible
+            break
```

Comment:
Instead of `break` here you could `return True, ...` right away, and then replace the rest of the code with the default case of `return False, ...`

Reply:
Nice one, done 👍